### DISCUSSION TITLE
FFMPEG vs MKVMERGE comparison for merging two videos and **some problems

### DISCUSSION THREAD

**Discussion by milos-99** [mifi/lossless-cut#1862] [General]
Hi,

I tried using 'Smart cut' feature few times on x254/x265  and it works for me on 1080p videos, but when I try to do the same with 4K anime videos I get kind of weird results. 

---------------------------------------------------------------------------------------------------------------------------------------------------

Results:
I play video in 'Media Player Classic' for first time and it works just fine like it was supposed to but then I let it play again (repeat mode) for second time, first keyframe (length of 3 seconds) is glitching =>grey screen, max pixelated.
Now, it is worth noting that this type of error is not happening in 'VLC' player.

![mpc-glitch](https://github.com/mifi/lossless-cut/assets/83019774/30f19dda-e25d-495e-9c1e-ac896de84e8d)

---------------------------------------------------------------------------------------------------------------------------------------------------

My experimentation:
Because I'll continue using 'MPC' I tried to fix this issue and most simple solution I came up with is:
1. to still use 'Smart cut' feature to extract that single 3 second keyframe separately
2. extract second segment without 'Smart cut' feature on keyframes
3. then merge both extracted videos into mkv file using mkvmerge
4. then use ffmpeg to do quick conversion from mkv to mp4
(but in same cases I mind skip last step and keep it as mkv)

---------------------------------------------------------------------------------------------------------------------------------------------------

Results from my experimentation:
I can now play concatenated video again (repeat mode) without it glitching.

---------------------------------------------------------------------------------------------------------------------------------------------------

That is mostly all I wanted to say, so you have one more reason to implement mkvmerge into Lossless-cut  if you ever considered it.
(under assumption if it does not get fixed inside ffmpeg)

---------------------------------------------------------------------------------------------------------------------------------------------------
**Random, extra thoughts:
1. Audio inside Lossless-cut is not working for codec: 'ac3 ac-3'
2. If possible add border or something for preview of tracks and for settings in general , so it is easer to follow each row in full screen mode.
3. Cutting mkv files works, but it always cuts extra keyframe at beginning 
4. When clicking next keyframe it knows to skip whole a lot of keyframes because they are not loaded, so if possible add background task for it.

**Comment by mifi** [comment]
for smart cut, see #126

for mkvmerge see https://github.com/mifi/lossless-cut/issues/895#issuecomment-950353355

> * Audio inside Lossless-cut is not working for codec: 'ac3 ac-3'

try convert to supported format

> * Cutting mkv files works, but it always cuts extra keyframe at beginning

#1216

> 4\. When clicking next keyframe it knows to skip whole a lot of keyframes because they are not loaded, so if possible add background task for it.

#563
